### PR TITLE
Iss244v2 Implements issue #244. Version 2

### PR DESCRIPTION
**Pull request by basaundi**
Changed the PR to include regular expression literals. Issue #244

### CODE REVIEW

File: simple-schema.js [hunk 1]
```diff
@@ -643,6 +673,22 @@ SimpleSchema.prototype.addValidator = SimpleSchema.prototype.validator = functio
...
 };
 
+// Used to match key parts.
+// * 1st group is the root key.
+// * 2nd group is the root key if it is a regular expression.
+// * 3rd group is the last key after a $.
+// * 4rd group is the last regular expression after a $.
+// * 5rd group is the leaf key.
+// * 6rd group is the last regular expression after the root.
+var KEY_REGEX = /^([^.\/]+|(\/[^\/]+\/))(?:\.\$\.([^.\/]+|(\/[^\/]+\/))|\.([^.\/]+|(\/[^\/]+\/)))*$/;
```
Comment (aldeed) [inline simple-schema.js]: This seems more complex than necessary. As far as I'm concerned, it's a regex key if it has at least two `/` characters anywhere in it. Am I missing something?

**Comment by aldeed** [comment]
Thanks, @basaundi. Generally this looks good. I will merge, test a bit, and release later this week.

File: simple-schema.js (hunk 1 above)
Comment (basaundi) [inline simple-schema.js]: Although I wouldn't recommend it, keynames can contain slashes so your proposal would interpret the following as a regular expression:
systems/wwp.etc/fstab: {}

**Comment by basaundi** [comment]
Thanks. As you can see, I've been busy lately but I still have interest on this feature.

**Pull request closed without merging**